src/Mvc/Mvc.Core/src/Infrastructure/ObjectResultExecutor.cs

```diff
@@ -18,16 +23,43 @@ namespace Microsoft.AspNetCore.Mvc.Infrastructure
     /// </summary>
     public class ObjectResultExecutor : IActionResultExecutor<ObjectResult>
     {
+        private delegate Task<object> ReadAsyncEnumerableDelegate(object value);
```

Comment:
Do we need this delegate type? Why not just `Func<object, Task<object>>`

Reply:
It was ugly to repeat it. But I can do a `using`

Reply:
There's no difference between this and a using. My point is that I find `Func<>` a lot more readable than custom delegate types. If you don't agree then feel free to ignore.

Reply:
I'd argue that the using doesn't create an extra type, so it's slightly better.